src/ingest_validation_tools/upload.py

```diff
@@ -371,7 +378,8 @@ def _constraint_checks(self, schema: SchemaVersion):
             "Authorization": f"Bearer {self.globus_token}",
             "Content-Type": "application/json",
         }
-        url = f"{self.app_context['constraints_url']}?match=True&order={CONSTRAINTS_CHECK_METHOD}"
+        params = urlencode({"match": True, "order": CONSTRAINTS_CHECK_METHOD})
+        url = urljoin(self.app_context["constraints_url"], f"?{params}")
         response = requests.post(url, headers=headers, data=data)
```

Comment:
You should be able to use the `params` parameter in the post call here:

https://requests.readthedocs.io/en/latest/user/quickstart/#passing-parameters-in-urls

So:
```
params = {"match": True, "order" CONSTRAINTS_CHECK_METHOD}
requests.post(self.app_context['constraints_url'], headers=headers, data=data, params=params)
```

Reply:
That's so much better!